### PR TITLE
Add negative and boundary-condition tests for Election lifecycle

### LINKED ISSUE #226: [Enhancement]: Missing negative and boundary test coverage for Election lifecycle functions
### Is there an existing issue for this?

- [x] I have searched the existing issues

## 📌 Issue Summary  
📌 Problem
The Election lifecycle contracts currently focus on happy-path tests. Several
invalid or boundary scenarios are either not tested or not explicitly asserted
with revert reasons.

This creates risk of silent regressions when modifying election logic.

Examples of missing or weak coverage:
• Calling lifecycle functions before election start
• Calling lifecycle functions after election end
• Invalid timestamps or durations during election creation
• Repeated or duplicate lifecycle actions
• Boundary conditions around start/end blocks

🎯 Proposal
Add explicit negative and boundary-condition tests to ensure:
• Correct reverts with expected error selectors
• Lifecycle invariants are enforced consistently
• Future refactors cannot accidentally relax constraints

🛠️ Scope
• Tests only (no contract logic changes unless a missing revert is discovered)
• Focus on Election / ElectionFactory lifecycle functions
• Use existing Hardhat setup (or Foundry if preferred and acceptable)

I’d like to work on this issue and submit a PR with focused test additions.

### Record

- [x] I have synced all my node versions as mentioned in the project
- [x] I am using the same version of npm as is the project
- [x] My current branch is in sync with the development branch
- [x] I want to work on this issue

### PR DESCRIPTION
**Pull request by harshkumarrai**
# Description

This PR adds explicit negative and boundary-condition test coverage for the Election contract lifecycle.

The new tests validate that lifecycle constraints are correctly enforced, preventing silent regressions when interacting with elections outside valid time windows or with invalid inputs.

This change is tests-only and does not modify any contract or production logic.
Fixes #226 

## Type of change

Please mark the options that are relevant.

- [ ] Updated UI/UX
- [ ] Improved the business logic of code
- [ ] Added new feature
- [x] Other

## Checklist:

- [x ] I have performed a self-review of my own code
- [x ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive test suite for Election contract lifecycle scenarios, including validation of voting eligibility, candidate management restrictions, and result availability checks with proper error handling.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: blockchain/test/lifecycle/Election.lifecycle.negative.test.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  await ethers.provider.send("evm_mine");
+
+  await expect(
+    election.connect(voter).userVote([0])
+  ).to.be.reverted;
+
+  await expect(
+    election.connect(voter).userVote([0])
+  ).to.be.reverted;
+});
```
[the remark this answers is not in the document]
Comment (harshkumarrai) [reply blockchain/test/lifecycle/Election.lifecycle.negative.test.js:69]: In this test, the election is intentionally initialized with a zero address ballot to keep the PR scoped to lifecycle boundary checks only. Because of this, the first userVote reverts before reaching the AlreadyVoted check.

I avoided introducing a mock ballot here to keep the PR minimal. If maintainers prefer strict AlreadyVoted validation, I am happy to add that in a follow-up PR with proper mocks.
